pkg/proxy/iptables/proxier.go

```diff
@@ -1128,24 +1128,22 @@ func (proxier *Proxier) syncProxyRules() {
 				)
 
 				destChain := svcXlbChain
-				// We have to SNAT packets to external IPs if externalTrafficPolicy is cluster.
+				// We have to SNAT packets to external IPs if externalTrafficPolicy is cluster
+				// and the traffic is NOT Local. Local traffic coming from Pods and Nodes will
+				// be always forwarded to the corresponding Service, so no need to SNAT
+				// If we can't differentiate the local traffic we always SNAT.
 				if !svcInfo.OnlyNodeLocalEndpoints() {
 					destChain = svcChain
-					writeLine(proxier.natRules, append(args, "-j", string(KubeMarkMasqChain))...)
+					// This masquerades off-cluster traffic to a External IP.
+					if proxier.localDetector.IsImplemented() {
+						writeLine(proxier.natRules, proxier.localDetector.JumpIfNotLocal(args, string(KubeMarkMasqChain))...)
```

Comment:
It's been a while since I've read the local detector code, but does this also account for traffic that would be caught with `--src-type LOCAL`?

Can we assume that externalIP _should_ bind locally in the same way `status.ingress[i].ip` (a.k.a loadbalancerIP) would? Loadbalancer IP has a check for `--src-type LOCAL` which accounts for host network traffic to external IPs. Should we add the same here?

Reply:
Ah nvm, this would end up jumping to `svcXLBChain` anyways which would have the `--src-type LOCAL` check if traffic was local.

Reply:
> does this also account for traffic that would be caught with `--src-type LOCAL`?

Curiously, NO.  It does not.  The "by CIDR" implementation will not catch hostNetwork pods.  Looking at how it is used today, it seems like we handle it specially ("// Next, redirect all src-type=LOCAL -> LB IP to the service chain") or is moot (MARK_MASQ does nothing).

I wonder if we should fix that more completely.  Awkward because there are some other rules with the same predicates (MARK_MASQ on the above case).